doc/pit/Platform_Integration_Test_high_level_design.md

```diff
@@ -0,0 +1,166 @@
+
+Besides auto generated test cases list, PIT also support single case or user specific case list as input.
+
+## 3.4 PIT component detailed
+Figure 3-2 shows the softawre component of PIT.
+![](images/PIT_conponents.png)
+
+### 3.4.1 Test cases list generator
+
+Test case list generator is used to generate a list of test cases, it parses users' input options, product specific platform configuration, and generate test case list from all test case database. This is used in the daily regression test and delivery test scenario.
+
+Each test case has a list of tags, denoting type, stage, etc. Current tag for stage includes: production, development, delivery. Tags for type include: auto, manual, utility. Other tags can be added, to provide extra information so that they can be grouped together for new scenario.
+
+### 3.4.2 Manual test options
+
+Alternatively, user can specify which case or a list of test cases to run. Optionally, users are able to specify how many times to run the cases. This is used in development stage for developers to debug/test a single feature, or be used in stress test scenario.
+
+### 3.4.3 Test result generator
+
+Test result is generated per case. There are 2 result formats:
+
+  1) Formatted strings, contains test case name, ID, result, type, etc. This standardized output is used in manufacturing process, so that output strings can be mapped to hardware/software/firmware/configuration fault easily.
+  2) Structured result, per current design, we're using JSON format, as it's commonly used and can be interfaced with users' automated acceptance checking system in the product delivery process.
+
+Besides test case result, a collective result is also generated, as a summary of the whole test. Similar to per test case result, it composes of standardized output string and structured data format (JSON).
+
+Figure 3-3 shows an example of final test result JSON.
+![](images/PIT_final_test_result.png)
+Figure 3-3
+
+### 3.4.4 Test case logic
+
+Figure 3-4 shows the test case logic and interaction
+![](images/PIT_test_case_logic.png)
+Figure 3-4
+
+Each test case has a configuration file, containing information such as test case name, description, type, and tag list.
+Each platform has a platform configuration file, containing platform specific (product specific) information such as CPU architecture, boot loader, CPU error monitor capability, peripheral related information (e.g. FAN number, PSU control options, etc.).
```

Comment:
Please clarify whether this platform configuration file is an existing file (like a [platform.json](https://github.com/sonic-net/sonic-buildimage/blob/master/device/dell/x86_64-dell_s6100_c2538-r0/platform.json)), or if a separate file needs to be created by platform vendors to support PIT.

If the latter, please provide an substantive example.

Reply:
> Please clarify whether this platform configuration file is an existing file (like a [platform.json](https://github.com/sonic-net/sonic-buildimage/blob/master/device/dell/x86_64-dell_s6100_c2538-r0/platform.json)), or if a separate file needs to be created by platform vendors to support PIT.
> 
> If the latter, please provide an substantive example.

Hi Jeff, I've added an example for these configuration files, it's vendor provided configuration.